astropy/modeling/polynomial.py

```diff
@@ -142,11 +142,7 @@ def _generate_coeff_names(self, ndim):
                 names.append('c{0}'.format(n))
         else:
             for i in range(self.degree + 1):
-                names.append('c{0}_{1}'.format(i, 0))
-            for i in range(1, self.degree + 1):
-                names.append('c{0}_{1}'.format(0, i))
-            for i in range(1, self.degree):
-                for j in range(1, self.degree):
+                for j in range(self.degree + 1):
                     if i + j < self.degree + 1:
                         names.append('c{0}_{1}'.format(i, j))
```

Comment:
@astrofrog It looks like the tests are failing because of the order of the coefficients. You need here and below
`...format(j, i))`